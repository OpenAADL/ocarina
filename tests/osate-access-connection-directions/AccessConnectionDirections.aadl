system needy
  features
    needsBus: requires bus access;
    needsData: requires data access;
end needy;

system giver
  features 
    givesBus: provides bus access ;
    givesData: provides data access ;
end giver;

system needyGiver
  features
    needsBus: requires bus access ;
    needsData: requires data access ;
    givesBus: provides bus access ;
    givesData: provides data access ;
end needyGiver;

system implementation needyGiver.i
  subcomponents
    b1: bus b;
    d1: data d;
    b2: bus b;
    d2: data d;
    
  connections
    -- good
    bus access b1 -> givesBus;
    -- good
    data access d1 -> givesData;
    
    -- bad
    bus access b2 -> needsBus;
    -- bad
    data access d2 -> needsData;
end needyGiver.i;

system implementation needyGiver.j
  subcomponents
    n: system needy;
    g: system giver;
    
  connections
    -- 
    bus access needsBus -> n.needsBus;
    bus access g.givesBus -> givesBus;
    data access needsData -> n.needsData;
    data access g.givesData -> givesData;
end needyGiver.j;


data D
end D;

bus B
end B;

system S
end S;

system implementation s.i
  subcomponents 
    n: system needy;
    g: system giver;
    
    badN: system needy;
    badG: system giver;
    
    b: bus B;
    d: data D;
    n2: system needy;
    
  connections
    -- good
    bus access g.givesBus -> n.needsBus;
    -- good
    data access g.givesData -> n.needsData;
    
    -- bad
    bus access badN.needsBus -> badG.givesBus;
    -- bad
    data access badN.needsData -> badG.givesData;
    
    --good
    bus access b -> n2.needsBus;
    -- good
    data access d -> n2.needsData;
end s.i;
