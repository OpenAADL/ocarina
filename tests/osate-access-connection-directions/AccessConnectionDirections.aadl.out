AccessConnectionDirections.aadl:35:05: b2 (subcomponent)  and needsBus (subcomponent access)  do not have compatible directions
AccessConnectionDirections.aadl:37:05: d2 (subcomponent)  and needsData (subcomponent access)  do not have compatible directions
AccessConnectionDirections.aadl:82:05: needsBus (subcomponent access)  and givesBus (subcomponent access)  do not have compatible directions
AccessConnectionDirections.aadl:84:05: needsData (subcomponent access)  and givesData (subcomponent access)  do not have compatible directions
Cannot analyze AADL specifications
